internal/gatewayapi/securitypolicy.go

```diff
@@ -316,23 +385,122 @@ func (t *Translator) buildCORS(policy *egv1a1.SecurityPolicy) *ir.CORS {
 				Suffix: &origin.Value,
 			})
 		case egv1a1.StringMatchRegularExpression:
+			if err := validateRegex(origin.Value); err != nil {
+				return nil, err // TODO zhaohuabing: also check regex in other places
+			}
 			allowOrigins = append(allowOrigins, &ir.StringMatch{
-				SafeRegex: &origin.Value, // TODO zhaohuabing: check if the value is a valid regex
+				SafeRegex: &origin.Value,
 			})
 		}
 	}
 
 	return &ir.CORS{
 		AllowOrigins:  allowOrigins,
-		AllowMethods:  policy.Spec.CORS.AllowMethods,
-		AllowHeaders:  policy.Spec.CORS.AllowHeaders,
-		ExposeHeaders: policy.Spec.CORS.ExposeHeaders,
-		MaxAge:        policy.Spec.CORS.MaxAge,
-	}
+		AllowMethods:  cors.AllowMethods,
+		AllowHeaders:  cors.AllowHeaders,
+		ExposeHeaders: cors.ExposeHeaders,
+		MaxAge:        cors.MaxAge,
+	}, nil
 }
 
-func (t *Translator) buildJWT(policy *egv1a1.SecurityPolicy) *ir.JWT {
+func (t *Translator) buildJWT(jwt *egv1a1.JWT) *ir.JWT {
 	return &ir.JWT{
-		Providers: policy.Spec.JWT.Providers,
+		Providers: jwt.Providers,
+	}
+}
+
+func (t *Translator) buildOIDC(
+	policy *egv1a1.SecurityPolicy,
+	resources *Resources) (*ir.OIDC, error) {
+	var (
+		oidc         = policy.Spec.OIDC
+		clientSecret *v1.Secret
+		err          error
+	)
+
+	from := crossNamespaceFrom{
+		group:     egv1a1.GroupName,
+		kind:      KindSecurityPolicy,
+		namespace: policy.Namespace,
 	}
+	if clientSecret, err = t.validateSecretRef(from, oidc.ClientSecret, resources); err != nil {
```

Comment:
are we validating anywhere that the `Secret` and `Policy` must be in the same namespace ?

Reply:
validateSecretRef checks three things:
  1. Does the secret reference have a valid Group and kind
  2. If the secret reference is a cross-namespace reference, is it permitted by any ReferenceGrant
  3. Does the secret exist

Reply:
would prefer stricter approach first of making sure `secret` and `policy` are in same namespace
and we can relax this later

Reply:
Gotcha